pkg/components/azure-arc-onboarding/component_test.go

```diff
@@ -0,0 +1,196 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azurearconboarding_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+
+	azurearconboarding "github.com/kinvolk/lokomotive/pkg/components/azure-arc-onboarding"
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func TestEmptyConfig(t *testing.T) {
+	c := azurearconboarding.NewConfig()
+
+	config := `component "azure-arc-onboarding" {}`
+
+	body, diagnostics := util.GetComponentBody(config, azurearconboarding.Name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Error getting component body: %v", diagnostics)
+	}
+
+	if diagnostics := c.LoadConfig(body, &hcl.EvalContext{}); !diagnostics.HasErrors() {
+		t.Fatal("Empty config should return errors as all the fields are required fields.")
+	}
+}
+
+func TestConfigSuccessWithArbitraryValues(t *testing.T) {
+	c := azurearconboarding.NewConfig()
+
+	config := `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+`
+
+	body, diagnostics := util.GetComponentBody(config, azurearconboarding.Name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Error getting component body: %v", diagnostics)
+	}
+
+	if diagnostics := c.LoadConfig(body, &hcl.EvalContext{}); diagnostics.HasErrors() {
+		t.Fatalf("Valid config should not return error, got: %s", diagnostics)
+	}
+
+	if _, err := c.RenderManifests(); err != nil {
+		t.Fatalf("Rendering manifests should not produce any error, got: %v", err)
+	}
+}
+
+func TestConfigWithEmptyValues(t *testing.T) {
+	c := azurearconboarding.NewConfig()
+
+	config := `
+component azure-arc-onboarding {
+  application_client_id = ""
+  tenant_id             = ""
+  application_password  = ""
+  resource_group        = ""
+  cluster_name          = ""
+}
+`
+
+	body, diagnostics := util.GetComponentBody(config, azurearconboarding.Name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Error getting component body: %v", diagnostics)
+	}
+
+	if diagnostics := c.LoadConfig(body, &hcl.EvalContext{}); !diagnostics.HasErrors() {
+		t.Fatalf("Config with empty fields should return error")
+	}
+}
+
+func TestConfig(t *testing.T) { //nolint:funlen
+	c := azurearconboarding.NewConfig()
+
+	tests := []struct {
+		desc    string
+		config  string
+		wantErr bool
+	}{
+		{
+			desc: "Valid config",
+			config: `
+component azure-arc-onboarding {
+  application_client_id = "test-application-id"
+  tenant_id             = "test-tenant-id"
+  application_password  = "test-application-password"
+  resource_group        = "test-resource-group"
+  cluster_name          = "foobar"
+}
+			`,
+		},
+		{
+			desc: "Empty config",
+			config: `
+component azure-arc-onboarding {}
```

Comment:
This test is the same as TestEmptyConfig.

Reply:
https://github.com/kinvolk/lokomotive/pull/1473#discussion_r654406137